Python/optimizer_bytecodes.c

```diff
@@ -446,7 +446,30 @@ dummy_func(void) {
     }
 
     op(_COMPARE_OP_INT, (left, right -- res)) {
-        res = sym_new_type(ctx, &PyBool_Type);
+        if (sym_is_const(ctx, left) && sym_is_const(ctx, right) &&
+        sym_matches_type(left, &PyLong_Type) && sym_matches_type(right, &PyLong_Type))
+        {
+            assert(PyLong_CheckExact(sym_get_const(ctx, left)));
+            assert(PyLong_CheckExact(sym_get_const(ctx, right)));
+            PyObject *tmp = PyObject_RichCompare(sym_get_const(ctx, left),
+                                                 sym_get_const(ctx, right),
+                                                 oparg >> 5);
+            if (tmp == NULL) {
+                goto error;
+            }
+            res = sym_new_const(ctx, tmp);
+            Py_DECREF(tmp);
+
+            if (_Py_IsImmortal(res)) {
+                REPLACE_OP(this_instr, _POP_TWO_LOAD_CONST_INLINE_BORROW, 0, (uintptr_t)tmp);
+            }
+            else {
+                res = sym_new_type(ctx, &PyBool_Type);
+            }
```

Comment:
`tmp` is always a `bool`, and `True` and `False` are both immortal, so `tmp` is always immortal. We can assert this for peace-of-mind, though.

```suggestion
            assert(PyBool_Check(tmp));
            assert(_Py_IsImmortal(tmp));
            REPLACE_OP(this_instr, _POP_TWO_LOAD_CONST_INLINE_BORROW, 0, (uintptr_t)tmp);
            res = sym_new_const(ctx, tmp);
            Py_DECREF(tmp);
```